index.js

```diff
@@ -84,28 +91,21 @@ module.exports = {
 
     var version   = require('./lib/version');
     var merge     = require('broccoli-merge-trees');
-    var addonTree = merge([version(), dir]);
-
-    if (process.env.EMBER_ENV === 'production') {
-      var strippedBuild = require('./lib/stripped-build');
-
-      // blacklist es6.modules so the modules are not compiled but simply the
-      // debug statements / features are stripped; this is taken from
-      // ember-cli-babel:
-      // https://github.com/babel/ember-cli-babel/blob/master/index.js#L71
-      var strippedAddon = strippedBuild('ember-data', addonTree, {
-        blacklist: ['es6.modules', 'useStrict']
-      });
 
-      return this._super.treeForAddon.call(this, strippedAddon);
-    }
-
-    return this._super.treeForAddon.call(this, addonTree);
+    return this._super.treeForAddon.call(this, merge([
+      version(),
+      dir
+    ]));
   },
 
   included: function(app) {
     this._super.included.apply(this, arguments);
 
+    if (process.env.EMBER_ENV === 'production') {
+      add(this.options.babel, 'blacklist', ['es6.modules', 'useStrict']);
```

Comment:
hmm, good catch. I admittedly have been lazy the last few updates here, sorry for not catching.